### PR TITLE
PP-2566 Adding initial documentation

### PR DESCRIPTION
**Pull request by kakumara**
1. Added a LICENSE, .gitignore files
2. Updated README to denote the potential initial API's of pay-apps.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 Crown Copyright (Government Digital Service)
```
Comment (markyjones) [inline LICENSE]: 2017

File: README.md [hunk 2]
```diff
@@ -1,2 +1,26 @@
 # pay-apps
-Dynamic Apps integrating with GOV.UK Pay   
+
+Offline apps integrating with GOV.UK Pay in Java(Dropwizard)
```
Comment (markyjones) [inline README.md]: Offline?

File: docs/api_specification.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+#### Request body description
+
+| Field                    | required | Description                                                      | Supported Values     |
+| ------------------------ |:--------:| ---------------------------------------------------------------- |----------------------|
+| `external_service_id`    |    X     | external service id of the Service as identified by adminusers.  |   |
+| `pay_api_token`          |    X     | valid api token for the gateway account of above service which this product takes payments for |  |
+| `description`            |    X     | Description of the product. This will be passed as the `description` when creating the charge | |
+| `price`                  |    X     | Price for the product in pence. This will be passed as the  `amount` when creating charge    | |
+| `return_url`             |          | Where to redirect to upon completion of a payment. If not provided, `pay-apps` will generate a default url to itself when creating a charge | |
```
Comment (markyjones) [inline docs/api_specification.md]: https only?

File: docs/api_specification.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+| `external_service_id`    | X              | external service id of the Service as identified by adminusers.  |   |
+| `description`            | X              | Description of the product |
+| `price`                  | X              | Price for the product in pence      |
+| `return_url`             |                | return url provided. _(not be available if it was not provided)_   |
+| `links.self`             | X              | self GET link to the product. |
+| `links.pay`              | X              | The link in `pay-apps-ui` where a charge for this product will be generated and redirected to GOV.UK Pay |
+
+---------------------------------------------------------------------------------------------------------------------------------------------------------
+
+## GET /v1/api/products/{productId}
```
Comment (markyjones) [inline docs/api_specification.md]: Would there be a need for a list too?

Comment (kakumara) [reply docs/api_specification.md]: 👍 yes .. for the list of product urls tab in selfservice. 
Will add one

File: docs/api_specification.md (hunk 3 above)
Comment (kakumara) [reply docs/api_specification.md]: 👍

File: LICENSE (hunk 1 above)
Comment (kakumara) [reply LICENSE]: 👍

File: README.md (hunk 2 above)
Comment (kakumara) [reply README.md]: not fully sure whats best to describe. Leaving it as just `apps` until we sort out the domain language next week.

**Comment by kakumara** [comment]
@markyjones addressed your review comments

File: docs/api_specification.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+### Request example
+
+```
+POST /v1/api/products
+Authorization: Bearer API_TOKEN
+Content-Type: application/json
+
+{
+    "external_service_id" :  "d76tt83563thcert8urihgiuer",
+    "pay_api_token" :        "5meusgv5ufclsbvde78mdf35bfdhnhm1307euq94kcf0lcqcqrovbjovib",
```
Comment (markyjones) [inline docs/api_specification.md]: @kakumara minor point for future is that I would avoid using "sane looking" keys in the code base as they stand out and to reviewers later in the day could cause alarm '*5meusgv5ufclsbv**EXAMPLE**qcqrovbjovib*' may be enough

**Pull request merged**